client/components/search/index.jsx

```diff
@@ -38,7 +38,8 @@ module.exports = React.createClass( {
 		onKeyDown: React.PropTypes.func,
 		disableAutocorrect: React.PropTypes.bool,
 		onBlur: React.PropTypes.func,
-		searching: React.PropTypes.bool
+		searching: React.PropTypes.bool,
+		forceCloseButton: React.PropTypes.bool
```

Comment:
Here's a gif that explains that flow that is broken by the missing `x`:

![add site](https://cloud.githubusercontent.com/assets/191598/11814720/5cdf08a2-a316-11e5-8b99-dfa5b2f2ef04.gif)

When I tap the Add button, the input slides in — but there's no way to dismiss it until you type something. This makes it really easy to "trap" someone in adding mode.